### PR TITLE
Backend: Updated getting active Mayor from API

### PR DESCRIPTION
**Pull request by j10a1n15**
## What
~~This Pull Request fixes the MayorAPI getting the wrong Mayor when the votes were hidden. This is caused by Hypixel for some reason also not showing the votes after the Election is over.~~ This makes the candidate map/list (I forgot) obsolete, will leave it in though.
[Discord](https://discord.com/channels/997079228510117908/1262551489071812648)

## Changelog Technical Details
+ Changed detecting active mayor to no longer use the number of votes. - j10a1n15

### CODE REVIEW

**Comment by j10a1n15** [comment]
Okay so Hypixel fixed this on their end by showing the election votes again. In theory this is obsolete, but imo this system is better for setting the active mayor so I'll keep this open

File: src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt [hunk 1]
```diff
@@ -195,20 +194,8 @@ object MayorAPI {
...
     }
 
-    private fun checkCurrentMayor() {
-        val nextMayorTime = calculateNextMayorTime()
-
-        // Check if it is still the mayor from the old SkyBlock year
-        currentMayor = candidates[nextMayorTime.toSkyBlockTime().year - 1]?.let {
-            if (it.name == lastMayor?.name) return
-
-            // TODO: Once Jerry is active, add the sub mayor perks in here
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt:205]: you lost the todo with those changes.

**Review by Thunderblade73 — APPROVED** [review]
Cleans up the code nicely. Only one small thing, you removed a not solved todo. (At least I assume it isn't solved, since nothing was changed related to jerry)

File: src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt (hunk 1 above)
Comment (j10a1n15) [reply src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt:205]: The todo actually got completed some prs ago using the Calendar menu

File: src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt [hunk 2]
```diff
@@ -195,20 +194,8 @@ object MayorAPI {
...
-            setAssumeMayorJson(it.name, it.perks)
-        }
-    }
-
-    private fun checkHypixelAPI() {
-        if (lastUpdate.passedSince() < 20.minutes || (currentMayor == Mayor.UNKNOWN && lastUpdate.passedSince() < 1.minutes)) return
+    private fun checkHypixelAPI(forceReload: Boolean = false) {
+        if (!forceReload && (lastUpdate.passedSince() < 20.minutes || (currentMayor == Mayor.UNKNOWN && lastUpdate.passedSince() < 1.minutes))) return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/data/MayorAPI.kt]: split this into two or more lines for readability and line length

**Review by hannibal002 — CHANGES_REQUESTED** [review]
looks fine and works fine

**Pull request merged**